### PR TITLE
Provides support for multiple manifest files.

### PR DESCRIPTION
**Pull request by dfar-io**
This PR will do the following:

1. Allow for scanning and presenting data for multiple manifest files within a repository.
2. Allows for manifest files to be found that do not exist in the repo root.

### CODE REVIEW

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (mrbiggred) [inline Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:124]: Why the `f.Substring(1)?` I'm sure there is a good reason but I'm just not smart enough to see it.

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (mrbiggred) [inline Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:120]: If we aren't doing anything with `files` it could just be returned:

```csharp
return Directory.GetFiles(clonedProjectRoot, ....);
```

**Review by mrbiggred — CHANGES_REQUESTED** [review]
It appears it will be hard to parse the text results from the Scan Result.  I think it would be better if the Lib returned the results as of an array of metric results.  That way the caller (CLI, Web, other clients) can format the results as they like.

I view Freshli Lib similar to a database.  The caller queries the database for rows of data.  The database does not format the data, just returns it.  It is up to the caller to format the data as needed (e.g. show a graph, summary, group by manifest file, etc).

As we use Freshli Lib we can add common filters/grouping of the data.  But these would just filter/group the data.  It would still be returned as an array of metric results or similar format.

Let me know if I misinterpreted the ScanResult class and other changes you made.

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (dfar-io) [reply Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:124]: I'll add a comment, but it clears the leading '/' in the filepath for the manifest file.

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (dfar-io) [reply Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:120]: Agreed

**Review by mrbiggred — CHANGES_REQUESTED** [review]
@dfar-io thank you for clarifying what the ScanResult returns.  If I understand correctly the each ScanResult class contains the manifest file and the scan data for that file.  The scan data is still returned as a MetricsResult array and can be accessed via the ScanResult class.

My new PR review request is to add some documentation and examples to the ReadMe.  Document what Lib will return and put an example or two of calling the Lib.

P.S. - Thanks for teaching me about .NET [records](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/records).  I learned something new today.

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (mrbiggred) [reply Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:125]: Would [GetRelativePath](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=net-5.0) work here?

File: Corgibytes.Freshli.Lib/ScanResult.cs
Comment (mrbiggred) [inline Corgibytes.Freshli.Lib/ScanResult.cs:9]: I don't think there are any tests for the `ToString` method.  Can you add one or two tests please and thank you.

File: README.md
Comment (mrbiggred) [inline README.md:84]: 👍

**Review by mrbiggred — CHANGES_REQUESTED** [review]
Two requests:

1) Can you use some .NET Path methods to parse the manifest files?

2) There are no tests for the `ScanResult.ToString` method.  Please add a test or two.

Other than the above the change looks good.  Thank you for updating the ReadMe file.

**Comment by mrbiggred** [comment]
Notes for myself.  I have fixed most of the approval tests but a couple still need to fixed.  Currently I am working on the `CpanfileDancer2` acceptance test and it fails in the `Corgibytes.Freshli.Lib\GitFileHistory.cs` at line 77.  It appears the problem is there is no matching date for some reason.  I'll have to figure out why there is no matching date and should update the code to not through an exception if the contents at the given date don't exist.

```
Corgibytes.Freshli.Lib.Test.Acceptance.CpanfileDancer2

System.InvalidOperationException: Sequence contains no matching element

System.InvalidOperationException
Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at System.Linq.Enumerable.Last[TSource](IEnumerable`1 source, Func`2 predicate)
   at Corgibytes.Freshli.Lib.GitFileHistory.GetKey(DateTime date) in C:\Repos\freshli\Corgibytes.Freshli.Lib\GitFileHistory.cs:line 77
   at Corgibytes.Freshli.Lib.GitFileHistory.ContentsAsOf(DateTime date) in C:\Repos\freshli\Corgibytes.Freshli.Lib\GitFileHistory.cs:line 62
   at Corgibytes.Freshli.Lib.Runner.ProcessAnalysisDate(String manifestFile, LibYearCalculator calculator, IFileHistory fileHistory, DateTime currentDate) in C:\Repos\freshli\Corgibytes.Freshli.Lib\Runner.cs:line 92
   at Corgibytes.Freshli.Lib.Runner.<>c__DisplayClass8_1.<ProcessManifestFiles>b__0(DateTime ad) in C:\Repos\freshli\Corgibytes.Freshli.Lib\Runner.cs:line 76
   at System.Linq.Enumerable.SelectEnumerableIterator`2.ToList()
   at Corgibytes.Freshli.Lib.Runner.ProcessManifestFiles(String analysisPath, DateTime asOf, FileHistoryFinder fileHistoryFinder) in C:\Repos\freshli\Corgibytes.Freshli.Lib\Runner.cs:line 75
   at Corgibytes.Freshli.Lib.Runner.Run(String analysisPath, DateTime asOf) in C:\Repos\freshli\Corgibytes.Freshli.Lib\Runner.cs:line 41
   at Corgibytes.Freshli.Lib.Test.Acceptance.CpanfileDancer2() in C:\Repos\freshli\Corgibytes.Freshli.Lib.Test\Acceptance.cs:line 109
```

**Comment by mrbiggred** [comment]
Approval tests are now fixed.  The solution was to write a custom NamerFactory that returns a generic OS.  There is the a [UniqueForOs](https://github.com/approvals/ApprovalTests.Net/blob/master/docs/ApprovalTests/EnvironmentSpecificTests.md) built into Approval Tests but it returns the specific Windows version, such as "Microsoft Windows 10 Professional" when all I want is "Windows".

No I need to review any above review comments and implement them.  Then I can hopefully get someone else to review this PR.

File: Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs
Comment (mrbiggred) [reply Corgibytes.Freshli.Lib/GitFileHistoryFinder.cs:125]: Changed to GetRelativePath.

**Review by mrbiggred — APPROVED** [review]
Approving the work done on my own change requests.  Will put this case up for review by someone else before merging it.

**Comment by mrbiggred** [comment]
Accidentally closed this PR.

**Review by dfar-io — COMMENTED** [review]
:+1:

**Pull request merged**
